client/my-sites/checkout/src/hooks/use-create-payment-complete-callback.tsx

```diff
@@ -76,13 +79,16 @@ export default function useCreatePaymentCompleteCallback( {
 	siteSlug: string | undefined;
 	sitelessCheckoutType?: SitelessCheckoutType;
 	checkoutFlow?: string;
+	connectAfterCheckout?: boolean;
+	adminUrl?: string;
+	fromSiteSlug?: string;
 } ): PaymentEventCallback {
 	const cartKey = useCartKey();
 	const { responseCart, reloadFromServer: reloadCart } = useShoppingCart( cartKey );
 	const reduxDispatch = useDispatch();
 	const siteId = useSelector( getSelectedSiteId );
 	const selectedSiteData = useSelector( getSelectedSite );
-	const adminUrl = selectedSiteData?.options?.admin_url;
+	const adminUrl = selectedSiteData?.options?.admin_url || wpAdminUrl;
```

Comment:
If `adminUrl` is re-defined here, does it need to be passed in the props?

Reply:
No, `const adminUrl` here is basically equal to: admin_url pulled from state, or if that is undefined/null, then equal to wpAdminUrl passed into the component from props, or if that is also undefined then adminUrl will be undefined.

I supplied more explanation [in the comment below](https://github.com/Automattic/wp-calypso/pull/82036#discussion_r1337605336).